<launch>
  <arg name = "id" default = "1010" />
  <arg name = "manual" default = "true" />
  <arg name = "thw" default = "1.0" />
  <arg name = "sensor2frontbumper" default = "0.15" />
  <arg name = "sensor2rearbumper" default = "0.11" />

  <arg name = "destination" default = "192.168.0.105" />
  <arg name = "port_send" default = "54654" />
  <arg name = "port_receive" default = "54754" /> 

  <arg name = "twist_type" default = "TwistStamped" />
  <arg name = "twist_cmd_topic" default = "/Car_000/speed_ref_2" />
  <arg name = "accel_cmd_topic" default = "/acc_cmd000_2" />
  
  <arg name = "curr_twist_topic" default = "/Car_000/curr_speed" />
  <arg name = "curr_pose_topic" default = "/vrpn_client_node/RigidBody001/pose" />

  <group ns = "Car_$(arg id)_2">
    <node pkg = "cacc_communicator" type = "car_sender" name = "car_sender">
      <param name = "destination" value = "$(arg destination)" />
      <param name = "port_send" value = "$(arg port_send)" />
      <param name = "id" value = "$(arg id)" />
      <param name = "manual" value = "$(arg manual)" />
      <param name = "thw" value = "$(arg thw)" />
      <param name = "sensor2frontbumper" value = "$(arg sensor2frontbumper)" />
      <param name = "sensor2rearbumper" value = "$(arg sensor2rearbumper)" />
      <param name = "curr_twist_topic" value = "$(arg curr_twist_topic)" />
      <param name = "curr_pose_topic" value = "$(arg curr_pose_topic)" />
      <param name = "twist_type" value = "$(arg twist_type)" />

      <remap from="/min_radar" to="/min_radar_2"/>
    </node>

    <node pkg = "cacc_communicator" type = "car_receiver" name = "car_receiver">
      <param name = "port_receive" value = "$(arg port_receive)" />
      <param name = "id" value = "$(arg id)" />
      <param name = "manual" value = "$(arg manual)" />
      <param name = "twist_cmd_topic" value = "$(arg twist_cmd_topic)" />
      <param name = "accel_cmd_topic" value = "$(arg accel_cmd_topic)" />

      <remap from="/com_status" to="/com_status_2"/>
    </node>
  </group>
</launch>

